be/src/exec/vectorized/sorting/compare_column.cpp

```diff
@@ -227,17 +232,27 @@ class ColumnTieBuilder final : public ColumnVisitorAdapter<ColumnTieBuilder> {
     template <typename T>
     Status do_visit(const vectorized::BinaryColumnBase<T>& column) {
         auto& data = column.get_data();
-        for (size_t i = 1; i < column.size(); i++) {
-            (*_tie)[i] &= SorterComparator<Slice>::compare(data[i - 1], data[i]) == 0;
+        TieIterator iterator(*_tie);
+        while (iterator.next()) {
+            int range_first = iterator.range_first;
+            int range_last = iterator.range_last;
+            for (size_t i = range_first + 1; i < range_last; i++) {
+                (*_tie)[i] &= SorterComparator<Slice>::compare(data[i - 1], data[i]) == 0;
+            }
         }
         return Status::OK();
     }
 
     template <typename T>
     Status do_visit(const vectorized::FixedLengthColumnBase<T>& column) {
         auto& data = column.get_data();
-        for (size_t i = 1; i < column.size(); i++) {
-            (*_tie)[i] &= SorterComparator<T>::compare(data[i - 1], data[i]) == 0;
+        TieIterator iterator(*_tie);
```

Comment:
Why change it ? Any performance benefit or regression ?

Reply:
reverted